KsApi/extensions/NSURLSession.swift

```diff
@@ -108,9 +130,8 @@ private let defaultSessionError =
 private let boundary = "k1ck574r73r154c0mp4ny"
 
 extension URLSession {
-  // Returns a producer that will execute the given upload once for each invocation of start().
-  fileprivate func rac_dataWithRequest(_ request: URLRequest, uploading file: URL, named name: String)
-    -> SignalProducer<(Data, URLResponse), Error> {
+  fileprivate func addFileUploadToRequest(_ request: URLRequest, uploading file: URL,
```

Comment:
A little bit of refactoring in here.

Reply:
nit: I find this name confusing; since it starts with a verb, I expected it to act on self instead of returning a thing. What do you think of `requestWithFileUpload` or something like that instead?